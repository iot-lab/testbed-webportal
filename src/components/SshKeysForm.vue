<template>
  <div>
    <ul class="nav nav-tabs" style="position: relative; top: 1px">
      <li v-for="(key, i) in keys" class="nav-item"> <a :href="'#tab_SSH'+i" :class="{ 'active': i === activeKey }" data-toggle="tab" @click="activeKey = i"
        class="nav-link">
        SSH key {{i+1}}
        <a @click.stop.prevent="delKey(i)" v-show="i === activeKey && keys.length > 1"><i class="fa fa-times-circle"></i></a> </a>
      </li>
      <li class="nav-item"><a @click="addKey" class="nav-link"><i class="fa fa-plus"></i></a>
      </li>
    </ul>
    <div class="tab-content" id="sshkeysTabContent">
      <div v-for="(key, i) in keys" class="tab-pane" :class="{ 'active': i === activeKey, 'show': i === 0 }"
      :id="'tab_SSH'+i">
        <div class="control-group">
          <div class="controls">
            <textarea v-model="keys[i]" class="form-control" :rows="rows"></textarea>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'SshKeys',

  props: {
    keys: {
      // List of SSH keys
      type: Array,
      default: () => [''],
    },
    rows: {
      // Number of rows in textarea
      type: Number,
      default: 4,
    },
  },

  data () {
    return {
      activeKey: 0,
    }
  },

  watch: {
    keys (newKeys, oldKeys) {
      if (this.activeKey >= this.keys.length || newKeys[0] !== oldKeys[0]) {
        this.activeKey = 0 // set first key active when keys list changes
      }
    },
  },

  methods: {
    addKey () {
      this.keys.push('')
      this.activeKey = this.keys.length - 1
    },
    delKey (index) {
      this.keys.splice(index, 1)
      this.activeKey = Math.min(this.activeKey, this.keys.length - 1)
    },
  },
}
</script>
