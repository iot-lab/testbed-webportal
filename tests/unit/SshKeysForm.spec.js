import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import SshKeysForm from '@/components/SshKeysForm'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('SshKeysForm.vue', () => {

  it('should match snapshot', () => {
    const wrapper = shallowMount(SshKeysForm, {localVue, router, propsData: {
      keys: ['key1', 'key2']
    }})
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should be able to add key', () => {
    const wrapper = shallowMount(SshKeysForm, {localVue, router, propsData: {
      keys: ['key1', 'key2']
    }})
    expect(wrapper.vm.keys).toHaveLength(2)
    wrapper.find('li:last-child a').trigger('click')
    expect(wrapper.vm.keys).toHaveLength(3)
  })

  it('should be able to del key', () => {
    const wrapper = shallowMount(SshKeysForm, {localVue, router, propsData: {
      keys: ['key1', 'key2']
    }})
    expect(wrapper.vm.keys).toHaveLength(2)
    wrapper.find('li a.active a').trigger('click')
    expect(wrapper.vm.keys).toHaveLength(1)
  })

})
