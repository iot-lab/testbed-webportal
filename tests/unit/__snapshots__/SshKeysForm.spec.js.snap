// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SshKeysForm.vue should match snapshot 1`] = `
<div>
  <ul
    class="nav nav-tabs"
    style="position: relative; top: 1px;"
  >
    <li
      class="nav-item"
    >
      <a
        class="nav-link active"
        data-toggle="tab"
        href="#tab_SSH0"
      >
        
      SSH key 1
      
        <a>
          <i
            class="fa fa-times-circle"
          />
        </a>
      </a>
    </li>
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
        data-toggle="tab"
        href="#tab_SSH1"
      >
        
      SSH key 2
      
        <a
          style="display: none;"
        >
          <i
            class="fa fa-times-circle"
          />
        </a>
      </a>
    </li>
     
    <li
      class="nav-item"
    >
      <a
        class="nav-link"
      >
        <i
          class="fa fa-plus"
        />
      </a>
    </li>
  </ul>
   
  <div
    class="tab-content"
    id="sshkeysTabContent"
  >
    <div
      class="tab-pane active show"
      id="tab_SSH0"
    >
      <div
        class="control-group"
      >
        <div
          class="controls"
        >
          <textarea
            class="form-control"
            rows="4"
          />
        </div>
      </div>
    </div>
    <div
      class="tab-pane"
      id="tab_SSH1"
    >
      <div
        class="control-group"
      >
        <div
          class="controls"
        >
          <textarea
            class="form-control"
            rows="4"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
